S/SLICOT/build_tarballs.jl

```diff
@@ -0,0 +1,109 @@
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "SLICOT"
+version = v"5.7.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com//SLICOT/SLICOT-Reference.git",
+              "7b96b6470ee0eaf75519a612d15d5e3e2857407d"),
+    ArchiveSource("https://github.com/Reference-LAPACK/lapack/archive/refs/tags/v3.8.0.tar.gz",
+              "deb22cc4a6120bff72621155a9917f485f96ef8319ac074a7afbc68aab88bcf6"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/SLICOT-Reference
+echo "cmake_minimum_required(VERSION 3.17)" >>CMakeLists.txt
+echo "project( cml )" >>CMakeLists.txt
+echo "enable_language( Fortran )" >>CMakeLists.txt
+echo "add_subdirectory( src )" >>CMakeLists.txt
+
+cd $WORKSPACE/srcdir/SLICOT-Reference/src
+cp ../../lapack-3.8.0/SRC/DEPRECATED/[dz]latzm.f .
+cp ../../lapack-3.8.0/SRC/DEPRECATED/dgegs.f .
+
+echo "set( SLICOT_SOURCE_FILES" >source_files.cmake
+ls *.f >>source_files.cmake
+echo ")" >>source_files.cmake
+
+LAB_SYMBOLS=(
+DASUM DAXPY DBDSQR DCABS1 DCOPY DDOT DGEBAK DGEBAL DGEBRD DGECON
+DGEEQU DGEES DGEEV DGEGS DGEHRD DGELQ2 DGELQF DGELS DGELSS DGELSY
+DGEMM DGEMV DGEQLF DGEQP3 DGEQR2 DGEQRF DGER DGERFS DGERQ2 DGERQF
+DGESC2 DGESV DGESVD DGESVX DGETC2 DGETRF DGETRI DGETRS DGGBAK DGGBAL
+DGGES DGGEV DHGEQZ DHSEQR DLABAD DLACN2 DLACPY DLADIV DLAEXC DLAG2
+DLAGV2 DLAHQR DLAIC1 DLALN2 DLAMC3 DLAMCH DLANGE DLANHS DLANSY DLANTR
+DLANV2 DLAPMT DLAPY2 DLAPY3 DLAQGE DLARF DLARFB DLARFG DLARFT DLARFX
+DLARNV DLARTG DLAS2 DLASCL DLASET DLASR DLASRT DLASSQ DLASV2 DLASY2
+DLATRS DLATZM DNRM2 DORG2R DORGBR DORGHR DORGQR DORGR2 DORGRQ DORM2R
+DORMBR DORMHR DORMLQ DORMQL DORMQR DORMR2 DORMRQ DORMRZ DPOCON DPOSV
+DPOTRF DPOTRS DPPTRF DPPTRI DPPTRS DPTTRF DPTTRS DROT DROTG DRSCL
+DSCAL DSPMV DSPR DSWAP DSYCON DSYEV DSYEVX DSYMM DSYMV DSYR2 DSYR2K
+DSYRK DSYSV DSYTRF DSYTRI DSYTRS DTGEX2 DTGEXC DTGSEN DTGSYL DTPMV
+DTRCON DTREVC DTREXC DTRMM DTRMV DTRSEN DTRSM DTRSV DTRSYL DTRTRI
+DTRTRS DTZRZF DZASUM DZNRM2 IDAMAX ILAENV IZAMAX LSAME XERBLA ZAXPY
+ZCOPY ZDRSCL ZDSCAL ZGECON ZGEES ZGEMM ZGEMV ZGEQP3 ZGEQRF ZGESV
+ZGESVD ZGETRF ZGETRI ZGETRS ZHGEQZ ZLACGV ZLACON ZLACP2 ZLACPY ZLADIV
+ZLAHQR ZLAIC1 ZLANGE ZLANHS ZLANTR ZLAPMT ZLARF ZLARFG ZLARNV ZLARTG
+ZLASCL ZLASET ZLASSQ ZLATRS ZLATZM ZROT ZSCAL ZSWAP ZTRSM ZTZRZF
+ZUNGQR ZUNMQR ZUNMRQ ZUNMRZ
+)
+
+
+echo "include(source_files.cmake)" >>CMakeLists.txt
+echo "add_library( slicot_shared SHARED \${SLICOT_SOURCE_FILES} )" >> CMakeLists.txt
+echo "target_link_libraries( slicot_shared \${LAPACK_blas_LIBRARIES})" >> CMakeLists.txt
+echo "set_target_properties( slicot_shared PROPERTIES OUTPUT_NAME slicot )" >> CMakeLists.txt
+echo "install( TARGETS slicot_shared LIBRARY DESTINATION lib )" >> CMakeLists.txt
+
+FFLAGS="${FFLAGS} -O2 -fPIC -ffixed-line-length-none -cpp"
+
+SYMBOL_DEFS=()
+if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
```

Comment:
Would you be OK with requiring Julia v1.7? If so, you can do
```suggestion
if [[ ${nbits} == 64 ]]; then
```
in any case this can't work on both Julia v1.6 and v1.7.

Reply:
@giordano I'm building a package depending on this which I'd like to work on  v1.6 and v1.7+.
IIUC the above condition is appropriate for Julia v1.6, so could we register a JLL as built here and then a later version with the v1.7 condition and ILP64 names for aarch64 as well? Or is there some other way to make Pkg handle architecture dependencies?

Reply:
I'd need to repeat the same comments as in https://github.com/JuliaPackaging/Yggdrasil/pull/4225#issuecomment-1010404532, especially the last paragraph:

> We need to find a solution for linking against OpenBLAS for both 1.6 _and_ 1.7, hopefully sooner than later, but at the moment I don't see a good _and_ future-proof solution.

Reply:
Also, since you are interested libblastrampoline, that's fully supported only in Julia v1.7. For example, MKL.jl won't use it anyway in Julia v1.6.

In general, targeting only Julia v1.7+ would solve many issues and let us move forward here quickly.

Reply:
Ok.  If there is demand for v1.6 support I'll deal with that later.

Reply:
To be clear, since you're now requiring Julia v1.7 you have to do
```suggestion
if [[ ${nbits} == 64 ]]; then
```